lte/gateway/c/oai/tasks/amf/amf_app_pdu_resource_setup_req_rel.cpp

```diff
@@ -0,0 +1,247 @@
+   * considering default or max bit rate.
+   * leveraged ambr calculation from qos_params_to_eps_qos and 24-501 spec used
+   */
+  ambr_calculation_pdu_session(smf_context, &dl_pdu_ambr, &ul_pdu_ambr);
+  ngap_pdu_ses_setup_req->ue_aggregate_maximum_bit_rate.dl = dl_pdu_ambr;
+  ngap_pdu_ses_setup_req->ue_aggregate_maximum_bit_rate.ul = ul_pdu_ambr;
+
+  // Hardcoded number of pdu sessions as 1
+  ngap_pdu_ses_setup_req->pduSessionResource_setup_list.no_of_items = 1;
+  ngap_pdu_ses_setup_req->pduSessionResource_setup_list.item[0].Pdu_Session_ID =
+      (Ngap_PDUSessionID_t)
+          smf_context->smf_proc_data.pdu_session_identity.pdu_session_id;
+
+  /* preparing for PDU_Session_Resource_Setup_Transfer.
+   * amf_pdu_ses_setup_transfer_req is the structure to be filled.
+   */
+  amf_pdu_ses_setup_transfer_req.pdu_aggregate_max_bit_rate.dl = dl_pdu_ambr;
+  amf_pdu_ses_setup_transfer_req.pdu_aggregate_max_bit_rate.ul = ul_pdu_ambr;
+
+  // UPF teid 4 octet and respective ip address are from SMF context
+  memcpy(
+      &amf_pdu_ses_setup_transfer_req.up_transport_layer_info.gtp_tnl.gtp_tied,
+      smf_context->gtp_tunnel_id.upf_gtp_teid, GNB_TEID_LEN);
+  amf_pdu_ses_setup_transfer_req.up_transport_layer_info.gtp_tnl
+      .endpoint_ip_address = blk2bstr(
+      &smf_context->gtp_tunnel_id.upf_gtp_teid_ip_addr, GNB_IPV4_ADDR_LEN);
+  amf_pdu_ses_setup_transfer_req.pdu_ip_type.pdn_type = IPv4;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_identifier =
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item.qos_flow_identifier;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.qos_characteristic.non_dynamic_5QI_desc.fiveQI =
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item
+          .qos_flow_level_qos_param.qos_characteristic.non_dynamic_5QI_desc
+          .fiveQI;
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.alloc_reten_priority.priority_level =
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item
+          .qos_flow_level_qos_param.alloc_reten_priority.priority_level;
+
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
+      .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_cap =
+      smf_context->pdu_resource_setup_req
+          .pdu_session_resource_setup_request_transfer
+          .qos_flow_setup_request_list.qos_flow_req_item
+          .qos_flow_level_qos_param.alloc_reten_priority.pre_emption_cap;
+
+  amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item
```

Comment:
Very verbose! Can be abstracted into a function/macro which takes as arguments 
- `amf_pdu_ses_setup_transfer_req.qos_flow_setup_request_list.qos_flow_req_item`
- `smf_context->pdu_resource_setup_req.pdu_session_resource_setup_request_transfer.qos_flow_setup_request_list.qos_flow_req_item`

Reply:
Fixed with memcopy(), LHS and RHS have same structure